src/babel.target.entry.plugin.ts

```diff
@@ -1,27 +1,32 @@
-import { compilation, Compiler, Plugin } from 'webpack'
-import Compilation = compilation.Compilation
-import Dependency = compilation.Dependency
-import NormalModuleFactory = compilation.NormalModuleFactory
+import { Compiler, EntryPlugin, Compilation, Dependency, NormalModuleFactory } from 'webpack'
 
 import { BabelTarget }                      from './babel-target'
-import { BabelTargetSingleEntryDependency } from './babel.target.single.entry.dependency'
 import { BabelTargetEntryDependency }       from './babel.target.entry.dependency'
 
-export abstract class BabelTargetEntryPlugin implements Plugin {
-
-  protected constructor(protected targets: BabelTarget[], protected context: string, protected name: string) {
-  }
+export class BabelTargetEntryPlugin implements EntryPlugin {
+  public constructor(protected targets: BabelTarget[], public context: string,
+    public entry: string, public options: EntryPlugin['options']) {}
```

Comment:
This is interesting - so does Webpack 5 basically support out of the box the kind of stuff I was doing to hack in support for multiple output targets? I'm wondering if there's more that could be done to strip things out and use the underlying Webpack functionality now 🤔

Reply:
No it doesn't. This is just a weird [typescript behavior](https://www.typescriptlang.org/play?#code/MYGwhgzhAEBiD29oG8BQ1rHgOwgFwCcBXYPeAgCgAciAjEAS2GjAC5psiBbWgUwICUKAL6pRqUJBgAhMAWgMuVELy69seGAiRoMWXIRJlKNek2i12nHvwA00U42ZsO3PoJFigA)sQ20oW3tkJ1EiYgAiAE9gBCQAdxj-JWxM3A0wZFMrZAA6XAIAalhwwppgUABuQKheKESxTmDQuwVIzkJY5IALQw1gNIDMrOAcvIqFYvwocoKq2lNFyrYOevGmlraQsK7naKgYiCEqUA0hUYzx7Nz8ypKFmZRCjeAtzJ2EVqh2g6OI69E7SOJCEA3BrYe7TJbVOhrWYfL7YfhojC8IA)